awx/main/models/events.py

```diff
@@ -520,13 +520,22 @@ def _update_host_summary_from_stats(self, hostnames):
                 (summary['host_id'], summary['id'])
                 for summary in JobHostSummary.objects.filter(job_id=job.id).values('id', 'host_id')
             )
+            updated_hosts = set()
             for h in all_hosts:
                 # if the hostname *shows up* in the playbook_on_stats event
                 if h.name in hostnames:
                     h.last_job_id = job.id
+                    updated_hosts.add(h)
                 if h.id in host_mapping:
                     h.last_job_host_summary_id = host_mapping[h.id]
-            Host.objects.bulk_update(all_hosts, ['last_job_id', 'last_job_host_summary_id'])
+                    updated_hosts.add(h)
+
+            def _chunks(x, n):
+                for i in range(0, len(x), n):
+                    yield x[i:i + n]
+
+            for chunk in _chunks(list(updated_hosts), 100):
```

Comment:
The other change is that I decided to split these updates into more reasonable chunks; I don't want to give up the performance benefits by calling `host.save()` per-Host, but it's also true that the list of hosts could be really large.